x-pack/plugins/ingest_pipelines/__jest__/client_integration/ingest_pipelines_filter_list.test.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { render, fireEvent } from '@testing-library/react';
+import React from 'react';
+import { IntlProvider } from '@kbn/i18n-react';
+import { act } from 'react-dom/test-utils';
+import {
+  FilterListButton,
+  Filters,
+  FilterName,
+} from '../../public/application/sections/pipelines_list/main';
+
+describe('FilterListButton', () => {
+  const mockData = [
+    { name: 'Pipeline 1', isManaged: true, processors: [] },
+    { name: 'Pipeline 2', isManaged: false, processors: [] },
+    { name: 'Pipeline 3', isManaged: true, processors: [] },
+    { name: 'Pipeline 4', isManaged: false, processors: [] },
+  ];
+
+  const mockFilters: Filters<FilterName> = {
+    managed: {
+      name: 'Managed pipelines',
+      checked: 'on',
+      handleFilter: jest.fn((allPipelines) => allPipelines.filter((p) => p.isManaged)),
+    },
+    notManaged: {
+      name: 'Not managed pipelines',
+      checked: 'on',
+      handleFilter: jest.fn((allPipelines) => allPipelines.filter((p) => !p.isManaged)),
+    },
+  };
+
+  const setup = () => {
+    const setFilters = jest.fn();
+    const handleChange = jest.fn().mockImplementation((filters) => {
+      setFilters(filters);
+    });
+
+    const { getByText } = render(
+      <IntlProvider locale="en" messages={{}}>
+        <FilterListButton filters={mockFilters} onChange={handleChange} />
+      </IntlProvider>
+    );
+
+    const viewButton = getByText('View');
+    act(() => {
+      fireEvent.click(viewButton);
+    });
+
+    const notManagedFilter = getByText('Not managed pipelines');
+    const managedFilter = getByText('Managed pipelines');
+
+    return {
+      setFilters,
+      viewButton,
+      notManagedFilter,
+      managedFilter,
+    };
+  };
+
+  it('should show only managed pipelines when only "Managed pipelines" filter is on', () => {
+    const { setFilters, notManagedFilter } = setup();
+
+    act(() => {
+      fireEvent.click(notManagedFilter);
+    });
+
+    expect(setFilters).toHaveBeenCalledWith(
+      expect.objectContaining({
+        managed: expect.objectContaining({ checked: 'on', handleFilter: expect.any(Function) }),
+        notManaged: expect.objectContaining({ checked: 'off', handleFilter: expect.any(Function) }),
+      })
+    );
+
+    const filteredPipelines = mockFilters.managed.handleFilter?.(mockData);
+    const onlyManagedPipelines = mockData.filter((pipeline) => pipeline.isManaged);
+
+    expect(filteredPipelines).toEqual(onlyManagedPipelines);
```

Comment:
I'm not sure I see the benefits of this test since you are mocking the `handleFilter` method. Similar question for the tests below. Can you explain?

I would recommend leveraging the existing [ingest_pipelinest_list.test.tsx file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/ingest_pipelines/__jest__/client_integration/ingest_pipelines_list.test.ts) to test the filter functionality, rather than having this standalone test file.